bin/validate_ami_bags.py

```diff
@@ -17,112 +14,143 @@ def _configure_logging(args):
+    parser.add_argument("--slow", action='store_false', help="Recalculate hashes (very slow)")
+    parser.add_argument("--metadata", action='store_true', help="Validate Excel metadata files")
     parser.add_argument('--log', help='The name of the log file')
     parser.add_argument('-q', '--quiet', action='store_true')
     return parser
 
-
-
-def main():
-    parser = _make_parser()
-    args = parser.parse_args()
-
-    bags = []
-
-    _configure_logging(args)
-
+def log_checks(args):
     checks = """Performing the following validations:
-    Checking 0xums,
+    Checking Oxums,
     Checking bag completeness,
     """
     if not args.slow:
         checks += "Recalculating hashes,\n"
-    checks += """Determing bag type,
+    checks += """Determining bag type,
     Checking directory structure,
     Checking filenames
     """
     if args.metadata:
-        checks += """Validating Excel/JSON metadata files against files
-    """
+        checks += "Validating Excel/JSON metadata files against files\n"
+    
     LOGGER.info(checks)
     LOGGER.info("""To interpret log messages:
     A WARNING means the bag is valid but has features that may need double-checking. It's most useful for legacy AMI bag conversion.
-    An ERROR means the bag is out of spec and cannot be ingested
+    An ERROR means the bag is out of spec and cannot be ingested.
     A CRITICAL means the script has failed. The bag may be in or out of spec.
     """)
 
+def process_directory(directory, args):
+    print("Now checking this directory:", directory)
+    directory_path = os.path.abspath(directory)
+    bags = []
+    for root, dirnames, filenames in os.walk(directory_path):
+        for dirname in dirnames:
+            if re.match(r'\d\d\d\d\d\d$', dirname):
+                bags.append(os.path.join(root, dirname))
+
+    if not bags:
+        LOGGER.info("No valid bag directories found in: {}".format(directory_path))
+        return {'directory': directory_path, 'summary': "No valid bag directories found"}
 
-    if args.directory:
-        for directory in args.directory:
-            directory_path = os.path.abspath(directory)
-            for root, dirnames, filenames in os.walk(directory_path):
-                for dirname in dirnames:
-                    if re.match(r'\d\d\d\d\d\d$', dirname):
-                        bags.append(os.path.join(root, dirname))
+    return process_bags(bags, args, directory_path)
 
-    if args.bagpath:
-        for bag in args.bagpath:
-            bags.append(os.path.abspath(bag))
+def process_single_bag(bagpath, args):
+    print("Now checking this bag:", bagpath)
+    bagpath = os.path.abspath(bagpath)
+    return process_bags([bagpath], args, bagpath)
+
+def process_bags(bags, args, directory_path):
 
-    LOGGER.info("Checking {} folder(s).".format(len(bags)))
+    # Log the number of bags or folders being processed
+    LOGGER.info("Checking {} folder(s)".format(len(bags)))
```

Comment:
moved this above `if args.quiet`... so that it always shows no matter what. also shortened it since the directory path is printed right above